### ISSUE TITLE
JavaScript: More testing section is hard to understand

### ISSUE THREAD

**Issue by MrLuckyCat** [TheOdinProject/curriculum#26432]
Maybe it's just me but I'm having a hard time understanding jest mocks. I've read every resource carefully but I'm still unable to understand them. Like how do I actually put them in practice? I'm trying to use them for Battleship project which is the last one in JavaScript course but I just couldn't understand. For example (the code is from jest: Mock Functions):
```js
//users.js
import axios from 'axios';

class Users {
  static all() {
    return axios.get('/users.json').then(resp => resp.data);
  }
}
```
```js
// users.test.js
import axios from 'axios';
import Users from './users';

jest.mock('axios');

test('should fetch users', () => {
  const users = [{name: 'Bob'}];
  const resp = {data: users};
  axios.get.mockResolvedValue(resp);

  // or you could use the following depending on your use case:
  // axios.get.mockImplementation(() => Promise.resolve(resp))

  return Users.all().then(data => expect(data).toEqual(users));
});
```

when I run the test both in js and ts, I get error axios.get.mockResolvedValue is not a function (I installed the package btw).

Like what is jest.mock() _actually_ doing? I don't get it and how to prevent this error? I tried googling but no luck.
Also how to mock for example, fetch api or console.log for testing? It's not clear how to mock such stuff with side effects.

This section should probably contain some resources to practice these skills or at least simplify mocking. 

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/node-path-javascript-more-testing

### Checks

- [x] I have thoroughly read and understand [The Odin Project Contributing Guide

**Comment by ManonLef** [comment]
I agree that mocking could be confusing. I'd love to see some ways to improve that content by providing an example that people can code along with in our own lesson. 

Perhaps someone more comfortable with testing from the @TheOdinProject/javascript has some ideas that we can open up for a contributor.

**Comment by bizarf** [comment]
This [video from Codete on Jest mocking](https://www.youtube.com/watch?v=OS5mVVM5vAg) might help. It's a more concise and better produced video than that Fun Fun Functions series that's listed as an assignment (those videos were too annoying).

There's also [this article on Medium which explains Jest mocking](https://medium.com/@rickhanlonii/understanding-jest-mocks-f0046c68e53c)

**Comment by Spaceflyy** [comment]
I would love to see an improvement to the mocking explanation. I found understanding the explanation about why we need testing and what a mock is fairly straight forward but more resources on how to actually implement them would be welcome
